Support/Mechanism/Models/chem-CH4-2step/mechanism.H

```diff
@@ -0,0 +1,2978 @@
+                 6.28411665e-10 * tc[2] - 8.66871176e-14 * tc[3];
+    // species 1: H2O
+    species[1] = +2.17691804e-03 - 3.28145036e-07 * tc[1] -
+                 2.91125961e-10 * tc[2] + 6.72803968e-14 * tc[3];
+    // species 2: CH4
+    species[2] = +1.33909467e-02 - 1.14657162e-05 * tc[1] +
+                 3.66877605e-09 * tc[2] - 4.07260920e-13 * tc[3];
+    // species 3: CO
+    species[3] = +2.06252743e-03 - 1.99765154e-06 * tc[1] +
+                 6.90159024e-10 * tc[2] - 8.14590864e-14 * tc[3];
+    // species 4: CO2
+    species[4] = +4.41437026e-03 - 4.42962808e-06 * tc[1] +
+                 1.57047056e-09 * tc[2] - 1.88833666e-13 * tc[3];
+    // species 5: N2
+    species[5] = +1.48797680e-03 - 1.13695200e-06 * tc[1] +
+                 3.02911140e-10 * tc[2] - 2.70134040e-14 * tc[3];
+  }
+}
+
+// compute an approx to the reaction Jacobian
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void
+aJacobian_precond(
+  amrex::Real* J, const amrex::Real* sc, const amrex::Real T, const int HP)
+{
+
+#if defined(PELE_COMPILE_AJACOBIAN) || !defined(AMREX_USE_HIP)
+  for (int i = 0; i < 49; i++) {
+    J[i] = 0.0;
+  }
+
+  amrex::Real wdot[6];
+  for (auto& val : wdot) {
+    val = 0.0;
+  }
+
+  const amrex::Real tc[5] = {
+    log(T), T, T * T, T * T * T, T * T * T * T}; // temperature cache
+  amrex::Real invT = 1.0 / tc[1];
+  amrex::Real invT2 = invT * invT;
+
+  // reference concentration: P_atm / (RT) in inverse mol/m^3
+  amrex::Real refC = 101325 / 8.31446 / T;
+  amrex::Real refCinv = 1.0 / refC;
+
+  // compute the mixture concentration
+  amrex::Real mixture = 0.0;
+  for (int k = 0; k < 6; ++k) {
+    mixture += sc[k];
+  }
+
+  // compute the Gibbs free energy
+  amrex::Real g_RT[6];
+  gibbs(g_RT, tc);
+
+  // compute the species enthalpy
+  amrex::Real h_RT[6];
+  speciesEnthalpy(h_RT, tc);
+
+  amrex::Real phi_f, k_f, k_r, phi_r, Kc, q, q_nocor, Corr, alpha;
+  amrex::Real dlnkfdT, dlnk0dT, dlnKcdT, dkrdT, dqdT;
+  amrex::Real dqdci, dcdc_fac, dqdc[6];
+  amrex::Real Pr, fPr, F, k_0, logPr;
+  amrex::Real logFcent, troe_c, troe_n, troePr_den, troePr, troe;
+  amrex::Real Fcent1, Fcent2, Fcent3, Fcent;
+  amrex::Real dlogFdc, dlogFdn, dlogFdcn_fac;
+  amrex::Real dlogPrdT, dlogfPrdT, dlogFdT, dlogFcentdT, dlogFdlogPr, dlnCorrdT;
+  const amrex::Real ln10 = log(10.0);
+  const amrex::Real log10e = 1.0 / log(10.0);
+  // reaction 0: 2 CH4 + 3 O2 => 2 CO + 4 H2O
+  // a non-third-body and non-pressure-fall-off reaction
+  // forward
+  phi_f = pow(sc[0], 3.000000) * (sc[2] * sc[2]);
+  k_f = 154500000 * exp(0.5 * tc[0] - (20075.8288822793) * invT);
+  dlnkfdT = 0.5 * invT + (20075.8288822793) * invT2;
+  // rate of progress
+  q = k_f * phi_f;
+  dqdT = dlnkfdT * k_f * phi_f;
+  // update wdot
+  wdot[0] -= 3 * q; // O2
+  wdot[1] += 4 * q; // H2O
+  wdot[2] -= 2 * q; // CH4
+  wdot[3] += 2 * q; // CO
+  // d()/d[O2]
+  dqdci = +k_f * 3.000000 * sc[0] * sc[0] * sc[2] * sc[2];
```

Comment:
I think this still needs to be updated for the FORD stuff. This appears to have been calculated using the stoichiometric coefficients as reaction orders rather than using the FORD reaction orders.

Reply:
@baperry2 Good catch! I think I got all of it but I could use another pair of eyes to make sure I caught everything